### DISCUSSION TITLE
Lien entre : Aiguillage et le formulaire

### DISCUSSION THREAD

**Discussion by GigMatt** [MTESSDev/FRW#67] [Annonces]
Bonjour, ma question est tout de même simple, mais importante. Quelle est la bonne pratique pour relier le formulaire d'aiguillage au formulaire. 

Avons-nous besoin de deux fichiers FRW distincts? Et comment pouvons-nous les lier ensemble?

Merci

**Comment by vinnyboii** [comment]
Bonjour,

Effectivement il vous faudra un formulaire distinct pour poser vos questions d'aiguillage et spécifier vers quels formulaires vous voulez rediriger selon les réponses saisies.

Dans le formulaire d'aiguillage, voici un exemple que vous pouvez mettre dans le bloc "config":
````
config:
  aiguillage:
    choix:
      # Placer les choix d'aiguillage en ordre de priorité afin d'exécuter le premier seulement  
      - vif: this.condition1()
        mode: redirectionFormulaire
        formulaire: form1
      - vif: this.condition2()
        mode: redirectionFormulaire
        formulaire: form2
  injecterJs: 
    method:  
      condition1:
        code: |
          () {
              if (this.form.champ1 === 'false' || this.estRegionFormulaire() === false){
                return true
              } 
              return false
          }
      condition2:
        code: |
          () {
              if (this.form.champ2=== 'true' && this.estRegionFormulaire()){
                return true
              }
              return false
          }
````
Dans cet exemple, nous avons défini deux méthodes dans le bloc "injecterJs" et nous les vérifions dans un v-if dans la section "aiguillage". Vous devrez les adapter à vos besoins et ajuster les paramètres "formulaire" pour spécifier vos codes de formulaires.

Une documentation plus approfondie sera disponible dans le P700U ultérieurement. En attendant, faites nous signe si vous avez des questions.

Merci!

**Reply by GigMatt** [reply]
Salut, il y a une ligne rouge sous aiguillage dans ces lignes. Je pense que c'est la cause que cela ne fonctionne pas de notre côté, et ce même en graduation.
config:
      aiguillage: (voir image) 
![image](https://github.com/MTESSDev/FRW/assets/126168415/7d44c341-3ca1-4adf-b18e-17939f49ef12)


Voici un autre indice de mon souci :
![image](https://github.com/MTESSDev/FRW/assets/126168415/d952f9c0-03b8-451d-8e2f-e298171629cf)
Merci de m'aider

**Reply by vinnyboii** [reply]
Bonjour,

Dans ton exemple, tu réfères à un fichier "form" dans le paramètre "formulaire".
Il faudrait seulement mettre le type de formulaire (nom du répertoire) dans ce champ.

Ce qui nous amène au second problème que j'ai constaté. Il y a 2 fichiers "form" dans votre répertoire RSP-079... 
Cela ne devrait pas arriver. Vous devriez avoir un fichier form par répertoire.
Pour que ton exemple fonctionne, il faudrait un répertoire distinct pour chaque formulaire. Soit un pour le formulaire d'aiguillage et 2 autres pour chacun des formulaires qui doivent être chainés après l'aiguillage.

Reviens moi si tu as d'autres questions.

**Reply by anisite** [reply]
Bonjour, un fix a été livré ce PM pour ajouter l'aiguillage au schéma, vous devriez donc en plus voir disparaître l'erreur dans l'outil du bac-à-sable. Merci de me tenir informé du résultat.
